src/urllib3/response.py

```diff
@@ -878,11 +878,7 @@ def read(
 
         data = self._raw_read(amt)
 
-        flush_decoder = False
-        if amt is None:
-            flush_decoder = True
-        elif amt != 0 and not data:
-            flush_decoder = True
+        flush_decoder = True if amt is None or (amt != 0 and not data) else False
```

Comment:
I like the change, `flush_decoder = amt is None or (amt != 0 and not data)` should be enough here